packages/optimizers/sharp/src/SharpOptimizer.js

```diff
@@ -0,0 +1,43 @@
+// @flow
+
+import {Optimizer} from '@parcel/plugin';
+import {blobToBuffer} from '@parcel/utils';
+import path from 'path';
+import sharp from 'sharp';
+
+const OPTIONS = {
+  jpeg: {
+    quality: 75,
+    mozjpeg: true,
+  },
+  png: {
+    palette: true,
+  },
+};
+
+const FORMATS = new Map<string, $Keys<typeof OPTIONS>>([
+  ['jpg', 'jpeg'],
+  ['jpeg', 'jpeg'],
+  ['png', 'png'],
+]);
+
+export default (new Optimizer({
+  async optimize({bundle, contents}) {
+    if (!bundle.env.shouldOptimize) {
+      return {contents};
+    }
+
+    const ext = path.extname(bundle.name).substr(1);
+    const format = FORMATS.get(ext);
+
+    if (!format) {
+      throw new Error(`Sharp does not support ${ext} images.`);
+    }
+
+    const optimized = await sharp(await blobToBuffer(contents))
```

Comment:
Not sure if we should add `.rotate()` to this as well to normalise the rotation metadata of images. We used to have an issue in the transformer where images would be rotated if they have a certain format, might also be caused by resizing not sure.

Reply:
Just seen #6444; looks like it shouldn't happen, but maybe worth having a test with the EXIF Orientation tag (more for the transformer though).

Reply:
Will expand on https://github.com/parcel-bundler/parcel/pull/6647.

Reply:
Turns out it _did_ happen, so I've solved it in https://github.com/parcel-bundler/parcel/pull/6624/commits/28fa471800b77b7e1f1e539e6c3bd1ae9a93b466.